### PR TITLE
Add TinyLlama model config

### PR DESCRIPTION
**Pull request by awaelchli**
Adds the TinyLlama 1.1B model config and instructions to download the checkpoint (trained up to 1T tokens so far). 
If you find this work useful, please cite the original authors: https://github.com/jzhang38/TinyLlama/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ Supports the following popular model checkpoints:
 | Meta AI [Code Llama](tutorials/download_code_llama.md)                         | 7B, 13B, 34B                       | [Rozière et al. 2023](https://arxiv.org/abs/2308.12950)                                          |
 | Microsoft Research [phi-1.5](tutorials/download_phi15.md)                      | 1.3B                               | [Li et al. 2023](https://arxiv.org/abs/2309.05463)                                               |
 | Mistral AI [Mistral](tutorials/download_mistral.md)                            | 7B                                 | [Mistral  website](https://mistral.ai/)                                                          |
+| StatNLP Research [TinyLlama](tutorials/download_mistral.md)                    | 1.1B                               | [Zhang et al. 2023](https://github.com/jzhang38/TinyLlama)
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply README.md:46]: It's written in their config file as the org: https://github.com/jzhang38/TinyLlama/blob/1f0da5366e37524e363a50dd7b8236086c00a5f3/lit_gpt/config.py#L249C2-L249C2

What is the meaning of the org field in the lit-gpt config dict?

File: lit_gpt/config.py [hunk 2]
```diff
@@ -1037,4 +1037,31 @@ def norm_class(self) -> Type:
         configs.append(copy)
 
 
+###################
+# StatNLP Research
+###################
+
+tiny_LLaMA = [
+    dict(
+        org="StatNLP-research",
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply lit_gpt/config.py:1046]: I don't know, I guessed org here means the name of the organization/research team that published the model

File: lit_gpt/config.py [hunk 3]
```diff
@@ -1037,4 +1037,31 @@ def norm_class(self) -> Type:
...
+        name="tiny-llama-1b",
```
Comment (carmocca) [inline lit_gpt/config.py:1047]: Using a generic model name won't easily work for our tests (see current failure). I would go with the safer option of listing the separate checkpoint configurations separately.

File: lit_gpt/config.py [hunk 4]
```diff
@@ -1037,4 +1037,31 @@ def norm_class(self) -> Type:
         configs.append(copy)
 
 
+###################
+# StatNLP Research
+###################
+
+tiny_LLaMA = [
```
Comment (carmocca) [inline lit_gpt/config.py:1044]: the original source is TinyLlama, so to be consistent this should be

```suggestion
tinyllama = [
```

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (carmocca) [reply README.md:46]: The HF org associated to the checkpoint

[the remark this answers is not in the document]
Comment (carmocca) [reply README.md:46]: Then lets ask the author! @jzhang38 should attribution be provided to statnlp or s-lab?

File: lit_gpt/config.py (hunk 3 above)
Comment (awaelchli) [reply lit_gpt/config.py:1046]: I find it unpractical that the names in our config are tied to HF. At least for pretraining it makes not a lot of sense.

File: lit_gpt/config.py [hunk 5]
```diff
@@ -1037,4 +1037,30 @@ def norm_class(self) -> Type:
         configs.append(copy)
 
 
+###################
+# StatNLP Research
```
Comment (carmocca) [inline lit_gpt/config.py:1041]:
```suggestion
# TinyLlama
```

File: lit_gpt/config.py [hunk 6]
```diff
@@ -1037,4 +1037,30 @@ def norm_class(self) -> Type:
...
+        block_size=2048,
+        vocab_size=32000,
+        padding_multiple=64,
+        n_layer=22,
+        n_head=32,
+        n_embd=2048,
+        rotary_percentage=1.0,
+        parallel_residual=False,
+        bias=False,
+        _norm_class="RMSNorm",  # original TinyLLama uses FusedRMSNorm
```
Comment (carmocca) [inline lit_gpt/config.py:1056]:
```suggestion
        _norm_class="RMSNorm",  # original TinyLlama uses FusedRMSNorm
```

File: tutorials/download_tinyllama.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+In order to use the TinyLLama 1.1B model checkpoint, which requires about 5 GB of disk space, download the weights and convert the checkpoint to the lit-gpt format:
+
+```bash
+pip install huggingface_hub
+
+python scripts/download.py --repo_id PY007/TinyLlama-1.1B-intermediate-step-480k-1T
+
+python scripts/convert_hf_checkpoint.py \
+    --checkpoint_dir checkpoints/PY007/TinyLlama-1.1B-intermediate-step-480k-1T \
+    --model_name tiny-llama-1b
```
Comment (carmocca) [inline tutorials/download_tinyllama.md:17]:
```suggestion
    --checkpoint_dir checkpoints/PY007/TinyLlama-1.1B-intermediate-step-480k-1T
```

**Review by carmocca — APPROVED** [review]

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (carmocca) [reply README.md]: Merging, but we can change this once you tell us your preference

**Pull request merged**